### DISCUSSION TITLE
Error! File /vercel/workpath0/node_modules/semver/preload.js does not exist.

### DISCUSSION THREAD

**Discussion by CarlosManotas** [vercel/vercel#5900] [Help]
I can't find anything related to this error on my vercel  deployl.

```
12:56:41.925 | Error: File /vercel/workpath0/node_modules/semver/preload.js does not exist.
-- | --
12:56:41.925 | at Job.emitDependency (/vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:37791)
12:56:41.925 | at /vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:38241
12:56:41.925 | at Array.map (<anonymous>)
12:56:41.925 | at Job.emitDependency (/vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:38021)
12:56:41.925 | at async /vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:38595
12:56:41.925 | at async Promise.all (index 0)
12:56:41.925 | at async Job.emitDependency (/vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:37992)
12:56:41.926 | at async /vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:38595
12:56:41.926 | at async Promise.all (index 3)
12:56:41.926 | at async Job.emitDependency (/vercel/afe367d3340d9eb2/.build-utils/.builder/node_modules/@vercelruntimes/next/dist/index.js:1:37992)
```
Everything works fine locally.

**Comment by mtt87** [comment]
Facing the same issue, `next build` works fine locally and "seems" like it also works on Vercel but then this error is displayed
![Screenshot 2021-02-26 at 17 51 50](https://user-images.githubusercontent.com/2150927/109336849-e1c3e180-785b-11eb-907b-6d7342c0f70d.png)

**Reply by hariria** [reply]
any updates on this @mtt87 ?

**Reply by mtt87** [reply]
I didn’t do any further investigation as there isn’t much I can do as I can’t reproduce it locally :(

**Reply by hariria** [reply]
got it no worries

**Reply by CarlosManotas** [reply]
In my case it was necessary to delete the node_modules folder and run again `npm install` or `yarn install`, it may not be your case but this is the theory.
The error seems to be caused by a missing package dependency in the package.json (maybe you forgot to save the package in package.json)

**Reply by mtt87** [reply]
I've tried of course to `rm -rf node_modules .next` and `yarn install` from scratch but everything works fine locally.
The problem is only on the Vercel deployment so it's harder to debug

**Reply by CarlosManotas** [reply]
Did you try run with the vercel CLI?

**Reply by hariria** [reply]
I tried removing all node modules and reinstalling and redeploying it and it worked, however, that shouldn't have made a difference since Vercel creates a fresh build every time i recommit. I would suggest just trying to redeploy using this method and seeing if that works

**Reply by GhostArrays** [reply]
I have been facing the same issue for the past week its really dumb

**Reply by mtt87** [reply]
Not sure how that would help? (talking about running it with the Vercel CLI)
Locally everything works fine with `next dev` and `next build`, not sure what the Vercel CLI would do differently

**Reply by mtt87** [reply]
> I tried removing all node modules and reinstalling and redeploying it and it worked, however, that shouldn't have made a difference since Vercel creates a fresh build every time i recommit. I would suggest just trying to redeploy using this method and seeing if that works

Already tried the `Redeploy` red button with `ignore build cache` but still facing the issue :( 

I'm confused that the error happens after the build finishes correctly so I'm not sure where that error is coming from but it feels like it's something in their infra.

A correct build has these final steps btw
```
11:49:19.935 | Done in 69.32s.
-- | --
11:49:21.313 | Traced Next.js serverless functions for external files in: 1348.320ms
11:49:21.596 | Compressed shared serverless function files: 283.159ms
11:49:21.645 | All serverless functions created in: 48.114ms
11:49:22.360 | Uploading build outputs...
11:49:30.689 | Build completed. Populating build cache...
11:50:09.714 | Uploading build cache [114.40 MB]...
11:50:11.702 | Build cache uploaded: 1987.473ms
11:50:11.931 | Done with "package.json"
```

So something in between there is broken.

FYI I recently removed the `target: "serverless"` from `next.config.js` as I was having other issues with that (even locally)

**Reply by dayon3** [reply]
> Did you try run with the vercel CLI?

Making a deployment with vercel CLI doesn't solve the issue.

![Screenshot-20210227184449-1069x316](https://user-images.githubusercontent.com/46526413/109392366-c674d700-792c-11eb-9cd8-2af396651d67.png)

**Reply by CarlosManotas** [reply]
@dayon3  Yes, I know, but after deleting the `node_modules` folder and rerunning locally` yarn install`, maybe you can try, I still don't know why this problem

**Reply by mtt87** [reply]
Problem solved by removing yarn.lock and regenerate it afterwards

**Reply by CarlosManotas** [reply]
in a moment I erased everything and then it worked fine, but if this is true let me smile  :)

**Reply by dayon3** [reply]
> in a moment I erased everything and then it worked fine, but if this is true let me smile :)

@mtt87 solution works like a charm. thank you

**Comment by mcsdevv** [comment]
Hi there, this was related to a problem with a Babel release that was fixed shortly after. Redeploying and removing the lockfile will resolve the issue as it will clear the build cache.

**Reply by mtt87** [reply]
Simply redeploying even with the checked `ignore build cache` didn't really work, tried multiple times.
What worked was to re-generate the yarn lock file
```
$ rm yarn.lock
$ yarn install
```

**Reply by villagra** [reply]
I'm facing the same issue since yesterday and unfortunately removing the yarn.lock is not working :( Any ideas?

**Reply by villagra** [reply]
I've updated next to the latest version 10.1.3 from 10.1.1 and that seems to have worked.

**Comment by luap2703** [comment]
I had the same error and first couldn't find out why. I found out that it was related to the use of **bcrypt** - Changing to bcryptjs solved this issue, as this package is 100% javascript.

Vercel also has an article about such issue here: https://vercel.com/guides/why-does-my-serverless-function-work-locally-but-not-when-deployed

I hope this helps someone facing a similar issue

**Reply by khachornchit** [reply]
I found a similar error and my error is
Error: File /vercel/path0/node_modules/simple-update-notifier/node_modules/semver/preload.js does not exist.

**Reply by jonsherrard** [reply]
@khachornchit did you fix this?

**Reply by jonsherrard** [reply]
I've got a feeling my issue is something to do with superjson
